kiwi/config/functions.sh

```diff
@@ -1137,4 +1137,65 @@ function baseQuoteFile {
     mv "${conf}" "${file}"
 }
 
+#======================================
+# baseVagrantSetup
+#--------------------------------------
+function baseVagrantSetup {
+    # insert the default insecure ssh key from here:
+    # https://github.com/hashicorp/vagrant/blob/master/keys/vagrant.pub
+    mkdir -p /home/vagrant/.ssh/
+    echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" > /home/vagrant/.ssh/authorized_keys
+    chmod 0600 /home/vagrant/.ssh/authorized_keys
+    chown -R vagrant:vagrant /home/vagrant/
+
+    # recommended ssh settings for vagrant boxes
+    echo "UseDNS no" >> /etc/ssh/sshd_config
+    echo "GSSAPIAuthentication no" >> /etc/ssh/sshd_config
+
+    # vagrant assumes that it can sudo without a password
+    # => add the vagrant user to the sudoers list
+    SUDOERS_LINE="vagrant ALL=(ALL) NOPASSWD: ALL"
+    if [ -d /etc/sudoers.d ]; then
+        echo "$SUDOERS_LINE" >| /etc/sudoers.d/vagrant
+        visudo -cf /etc/sudoers.d/vagrant
+        chmod 0440 /etc/sudoers.d/vagrant
+    else
+        echo "$SUDOERS_LINE" >> /etc/sudoers
+        visudo -cf /etc/sudoers
+    fi
+
+    # the default shared folder
+    mkdir -p /vagrant
+    chown -R vagrant:vagrant /vagrant
+
+    # SSH service
+    baseInsertService sshd
+}
+
+#======================================
+# suseVagrantSetup
+#--------------------------------------
+function suseVagrantSetup {
+    baseVagrantSetup
+
+    # start vboxsf service only if the guest tools are present
+    if Rpm -q virtualbox-guest-tools 2> /dev/null; then
+        echo vboxsf > /etc/modules-load.d/vboxsf.conf
+    fi
+
+    # drop any network udev rules for libvirt, so that the networks are called
+    # ethX
+    # this is not required for Virtualbox as it handles networking differently
+    # and doesn't need this hack
+    if [ "${kiwi_profiles}" != "virtualbox" ]; then
+        Rm -f /etc/udev/rules.d/*-net.rules
```

Comment:
`net.ifnames=0` as kernel parameter might be better than deleting udev rules. IMO `rm` calls like this aren't a good idea as they'll just come back after the next update installation.

Reply:
Hm, after digging into this a bit I've found out that I already have this setting in the image description and it was (probably) not sufficient.

You're right concerning updates, but that's not really a typical workflow for vagrant boxes. These tend to get redeployed instead of getting updates.

Reply:
I agree to what Fabian said, deleting udev rules that way is unexpected

Reply:
Is there a reason we don't just use NetworkManager and have it autoconfigure network stuff inside?

Reply:
as I said there are several ways to setup a network and every distro has their own concepts. I think network setup is out of scope and should be an individual tasks of the author of the image description. With regards to the vagrant setup a network is probably mandatory so this aspect and maybe an example should be ok as documentation